shed/tests/test_streams.py

```diff
@@ -454,6 +454,25 @@ def f(img1):
         assert n in assert_docs
 
 
+def test_filter_full_header(exp_db, start_uid1):
+    source = Stream()
+
+    def f(docs):
+        d = docs[0]
+        return d['sample_name'] != 'hi'
```

Comment:
We do have control over exp_db it's in test/conftest.py all the samples are 'hi'. But you are right, I should really duplicate all the tests so that we can run them against the run engine and databroker.